actionpack/lib/action_dispatch/system_testing/test_helpers/screenshot_helper.rb

```diff
@@ -42,7 +42,10 @@ def take_screenshot(html: false, screenshot: nil)
         #
         # +take_failed_screenshot+ is called during system test teardown.
         def take_failed_screenshot
-          take_screenshot if failed? && supports_screenshot? && Capybara::Session.instance_created?
+          return unless failed? && supports_screenshot? && Capybara::Session.instance_created?
+
+          take_screenshot
+          metadata[:failure_screenshot_path] = absolute_image_path
```

Comment:
Bumping minitest in the gemfile only does it for the Rails CI builds but not applications. We locked an older Ruby to an older minitest becasue of the kwargs changes. You can see this is causing the tests to fail for that one here: https://buildkite.com/rails/rails/builds/98449#0189b14e-eeaf-4d2d-9b0f-5382c140fe23/1062-1226

We can't set this or test it unless `metadata` is defined. We also don't want to force applications to upgrade to 5.19 for this feature.

Reply:
I'm a big fan of your work @eileencodes! I feel like talking to a real life action heroine. 😊

Back to reality - so you're saying we need to check if `metadata` is defined here before doing anything with it? 🤔

Is that preferable to backporting the functionality to older versions of `minitest`? Not sure if the the maintainer is open to the idea, but I could try asking.

Reply:
I added a check for the existence of `metadata`.

To make the tests work, I skipped the test dynamically if `metadata` is not defined on `Minitest::Runnable`. This leaks a little bit of implementation into the test, but I wasn't sure how else I could make the test work for older versions of `minitest`.

I also contemplated adding an additional test case that checks how the helper behaves in a scenario when it's running with an older version of `minitest` which doesn't have `metadata` defined. i.e. to make sure that it doesn't raise an error. But in order to do that, I'd need to undefine `metadata` then redefine it after running the test. I opted against this, because I deemed the risk of the test leaking bigger than the risk of the error that it's trying to prevent.